pkg/operators/ebpf/stats.go

```diff
@@ -0,0 +1,535 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ebpfoperator
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/spf13/viper"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/bpfstats"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	metadatav1 "github.com/inspektor-gadget/inspektor-gadget/pkg/metadata/v1"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	ParamGadgetStatisticsInterval = "statistics-interval"
+	ParamGadgetStatisticsCount    = "statistics-count"
+	EmitStatsAnn                  = "emitstats"
+	ParamProgramStats             = "programs"
+	StatsDSName                   = "bpfstats"
+	EnableBPFStatsParam           = "enable-bpfstats"
+)
+
+func (i *ebpfOperator) GlobalParams() api.Params {
+	return api.Params{
+		{
+			Key:          EnableBPFStatsParam,
+			Description:  "Enable capturing eBPF stats",
+			DefaultValue: "false",
+			TypeHint:     api.TypeBool,
+			Title:        "Enable eBPF stats",
+		},
+	}
+}
+
+func (i *ebpfOperator) Init(params *params.Params) error {
+	if !params.Get(EnableBPFStatsParam).AsBool() {
+		return nil
+	}
+
+	i.gadgetObjs = make(map[operators.GadgetContext]gadgetObjs)
+
+	// Enable stats collection
+	// TODO: when to disable it?
+	if err := bpfstats.EnableBPFStats(); err != nil {
+		return fmt.Errorf("enabling bpf stats: %w", err)
+	}
+
+	return nil
+}
+
+func (i *ebpfOperator) InstantiateDataOperator(
```

Comment:
I think the idea of making the ebpf op also a data operator is brilliant! Bravo!